Errors while generating `wafv2-controller` for ACK runtime `v0.59.1`, code-generator `v0.59.1`

### Errors while generating `wafv2-controller` for ACK runtime `v0.59.1`, code-generator `v0.59.1`

#### stdout for `make build-controller`:
```
building ack-generate ... ok.
==== building wafv2-controller ====
Copying common custom resource definitions into wafv2
Building Kubernetes API objects for wafv2
```

#### stderr for `make build-controller`:
```
Error: resource "WebACL", custom nested field "Rules.Statement.AndStatement": member "AndStatement" already exists in shape of type "structure"
make: *** [Makefile:36: build-controller] Error 2
```


#### Steps for closing this issue:
1. Update wafv2-controller `go.mod` file to update `aws-controllers-k8s/runtime` to v0.59.1
2. Run `go mod tidy` for wafv2-controller
3. Successfully generate the service controller locally using latest release of
`aws-controllers-k8s/code-generator`
4. Run `make test` for wafv2-controller
5. Run `make kind-test` from `aws-controllers-k8s/test-infra` repository for
 wafv2-controller
6. When kind test is successful, Create new pull request for wafv2-controller
7. Mention this issue number in the pull request
8. When the above PR is merged, close this issue

For more guidance, [create an issue](https://github.com/aws-controllers-k8s/community/issues)
or [reach out on Slack](https://github.com/aws-controllers-k8s/community#help--feedback).

--- a/apis/v1alpha1/ack-generate-metadata.yaml
+++ b/apis/v1alpha1/ack-generate-metadata.yaml
@@ -1,8 +1,8 @@
 ack_generate_info:
-  build_date: "2026-04-22T18:21:26Z"
-  build_hash: c55e8227b1ff5ac6a8b8e1421a2f2076fa6a77bd
+  build_date: "2026-05-27T16:42:38Z"
+  build_hash: d4db016091e4579dcbbfec59495c3856581a0b3f
   go_version: go1.26.2
-  version: v0.58.1
+  version: v0.59.1-3-gd4db016
 api_directory_checksum: b76e1cbc62fb6cc6a9c5bd0fd0a1c31fd6c701c1
 api_version: v1alpha1
 aws_sdk_go_version: v1.32.6

--- a/config/crd/bases/wafv2.services.k8s.aws_ipsets.yaml
+++ b/config/crd/bases/wafv2.services.k8s.aws_ipsets.yaml
@@ -173,6 +173,10 @@ spec:
                       OwnerAccountID is the AWS Account ID of the account that owns the
                       backend AWS service API resource.
                     type: string
+                  partition:
+                    description: Partition is the AWS partition in which the resource
+                      exists or will exist
+                    type: string
                   region:
                     description: Region is the AWS region in which the resource exists
                       or will exist.

--- a/config/crd/bases/wafv2.services.k8s.aws_rulegroups.yaml
+++ b/config/crd/bases/wafv2.services.k8s.aws_rulegroups.yaml
@@ -3797,6 +3797,10 @@ spec:
                       OwnerAccountID is the AWS Account ID of the account that owns the
                       backend AWS service API resource.
                     type: string
+                  partition:
+                    description: Partition is the AWS partition in which the resource
+                      exists or will exist
+                    type: string
                   region:
                     description: Region is the AWS region in which the resource exists
                       or will exist.

--- a/config/crd/bases/wafv2.services.k8s.aws_webacls.yaml
+++ b/config/crd/bases/wafv2.services.k8s.aws_webacls.yaml
@@ -4336,6 +4336,10 @@ spec:
                       OwnerAccountID is the AWS Account ID of the account that owns the
                       backend AWS service API resource.
                     type: string
+                  partition:
+                    description: Partition is the AWS partition in which the resource
+                      exists or will exist
+                    type: string
                   region:
                     description: Region is the AWS region in which the resource exists
                       or will exist.

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/aws-controllers-k8s/wafv2-controller
 go 1.25.0
 
 require (
-	github.com/aws-controllers-k8s/runtime v0.58.0
+	github.com/aws-controllers-k8s/runtime v0.59.1
 	github.com/aws/aws-sdk-go v1.51.21
 	github.com/aws/aws-sdk-go-v2 v1.34.0
 	github.com/aws/aws-sdk-go-v2/service/wafv2 v1.55.11
@@ -51,7 +51,7 @@ require (
 	github.com/josharian/intern v1.0.0 // indirect
 	github.com/json-iterator/go v1.1.12 // indirect
 	github.com/mailru/easyjson v0.7.7 // indirect
-	github.com/micahhausler/aws-iam-policy v0.4.4 // indirect
+	github.com/micahhausler/aws-iam-policy v0.4.5-0.20260511184658-411e29b8ffd2 // indirect
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.3-0.20250322232337-35a7c28c31ee // indirect
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect

--- a/helm/crds/wafv2.services.k8s.aws_ipsets.yaml
+++ b/helm/crds/wafv2.services.k8s.aws_ipsets.yaml
@@ -173,6 +173,10 @@ spec:
                       OwnerAccountID is the AWS Account ID of the account that owns the
                       backend AWS service API resource.
                     type: string
+                  partition:
+                    description: Partition is the AWS partition in which the resource
+                      exists or will exist
+                    type: string
                   region:
                     description: Region is the AWS region in which the resource exists
                       or will exist.

--- a/helm/crds/wafv2.services.k8s.aws_rulegroups.yaml
+++ b/helm/crds/wafv2.services.k8s.aws_rulegroups.yaml
@@ -3797,6 +3797,10 @@ spec:
                       OwnerAccountID is the AWS Account ID of the account that owns the
                       backend AWS service API resource.
                     type: string
+                  partition:
+                    description: Partition is the AWS partition in which the resource
+                      exists or will exist
+                    type: string
                   region:
                     description: Region is the AWS region in which the resource exists
                       or will exist.

--- a/helm/crds/wafv2.services.k8s.aws_webacls.yaml
+++ b/helm/crds/wafv2.services.k8s.aws_webacls.yaml
@@ -4336,6 +4336,10 @@ spec:
                       OwnerAccountID is the AWS Account ID of the account that owns the
                       backend AWS service API resource.
                     type: string
+                  partition:
+                    description: Partition is the AWS partition in which the resource
+                      exists or will exist
+                    type: string
                   region:
                     description: Region is the AWS region in which the resource exists
                       or will exist.

--- a/pkg/resource/ip_set/identifiers.go
+++ b/pkg/resource/ip_set/identifiers.go
@@ -53,3 +53,12 @@ func (ri *resourceIdentifiers) Region() *ackv1alpha1.AWSRegion {
 	}
 	return nil
 }
+
+// Partition returns the AWS partition in which the reosurce exists, or
+// nil if this information is not known.
+func (ri *resourceIdentifiers) Partition() *ackv1alpha1.AWSPartition {
+	if ri.meta != nil {
+		return ri.meta.Partition
+	}
+	return nil
+}

--- a/pkg/resource/ip_set/manager.go
+++ b/pkg/resource/ip_set/manager.go
@@ -75,6 +75,8 @@ type resourceManager struct {
 	awsAccountID ackv1alpha1.AWSAccountID
 	// The AWS Region that this resource manager targets
 	awsRegion ackv1alpha1.AWSRegion
+	// The AWS Partition that this resource manager targets
+	awsPartition ackv1alpha1.AWSPartition
 	// sdk is a pointer to the AWS service API client exposed by the
 	// aws-sdk-go-v2/services/{alias} package.
 	sdkapi *svcsdk.Client
@@ -193,7 +195,8 @@ func (rm *resourceManager) Delete(
 // name for the resource
 func (rm *resourceManager) ARNFromName(name string) string {
 	return fmt.Sprintf(
-		"arn:aws:wafv2:%s:%s:%s",
+		"arn:%s:wafv2:%s:%s:%s",
+		rm.awsPartition,
 		rm.awsRegion,
 		rm.awsAccountID,
 		name,
@@ -345,6 +348,7 @@ func newResourceManager(
 		rr:           rr,
 		awsAccountID: id,
 		awsRegion:    region,
+		awsPartition: ackv1alpha1.AWSPartition(cfg.Partition),
 		sdkapi:       svcsdk.NewFromConfig(clientcfg),
 	}, nil
 }

--- a/pkg/resource/ip_set/sdk.go
+++ b/pkg/resource/ip_set/sdk.go
@@ -379,6 +379,9 @@ func (rm *resourceManager) setStatusDefaults(
 	if ko.Status.ACKResourceMetadata.Region == nil {
 		ko.Status.ACKResourceMetadata.Region = &rm.awsRegion
 	}
+	if ko.Status.ACKResourceMetadata.Partition == nil {
+		ko.Status.ACKResourceMetadata.Partition = &rm.awsPartition
+	}
 	if ko.Status.ACKResourceMetadata.OwnerAccountID == nil {
 		ko.Status.ACKResourceMetadata.OwnerAccountID = &rm.awsAccountID
 	}

--- a/pkg/resource/rule_group/identifiers.go
+++ b/pkg/resource/rule_group/identifiers.go
@@ -53,3 +53,12 @@ func (ri *resourceIdentifiers) Region() *ackv1alpha1.AWSRegion {
 	}
 	return nil
 }
+
+// Partition returns the AWS partition in which the reosurce exists, or
+// nil if this information is not known.
+func (ri *resourceIdentifiers) Partition() *ackv1alpha1.AWSPartition {
+	if ri.meta != nil {
+		return ri.meta.Partition
+	}
+	return nil
+}

--- a/pkg/resource/rule_group/manager.go
+++ b/pkg/resource/rule_group/manager.go
@@ -75,6 +75,8 @@ type resourceManager struct {
 	awsAccountID ackv1alpha1.AWSAccountID
 	// The AWS Region that this resource manager targets
 	awsRegion ackv1alpha1.AWSRegion
+	// The AWS Partition that this resource manager targets
+	awsPartition ackv1alpha1.AWSPartition
 	// sdk is a pointer to the AWS service API client exposed by the
 	// aws-sdk-go-v2/services/{alias} package.
 	sdkapi *svcsdk.Client
@@ -193,7 +195,8 @@ func (rm *resourceManager) Delete(
 // name for the resource
 func (rm *resourceManager) ARNFromName(name string) string {
 	return fmt.Sprintf(
-		"arn:aws:wafv2:%s:%s:%s",
+		"arn:%s:wafv2:%s:%s:%s",
+		rm.awsPartition,
 		rm.awsRegion,
 		rm.awsAccountID,
 		name,
@@ -373,6 +376,7 @@ func newResourceManager(
 		rr:           rr,
 		awsAccountID: id,
 		awsRegion:    region,
+		awsPartition: ackv1alpha1.AWSPartition(cfg.Partition),
 		sdkapi:       svcsdk.NewFromConfig(clientcfg),
 	}, nil
 }

--- a/pkg/resource/rule_group/sdk.go
+++ b/pkg/resource/rule_group/sdk.go
@@ -6243,6 +6243,9 @@ func (rm *resourceManager) setStatusDefaults(
 	if ko.Status.ACKResourceMetadata.Region == nil {
 		ko.Status.ACKResourceMetadata.Region = &rm.awsRegion
 	}
+	if ko.Status.ACKResourceMetadata.Partition == nil {
+		ko.Status.ACKResourceMetadata.Partition = &rm.awsPartition
+	}
 	if ko.Status.ACKResourceMetadata.OwnerAccountID == nil {
 		ko.Status.ACKResourceMetadata.OwnerAccountID = &rm.awsAccountID
 	}

--- a/pkg/resource/web_acl/identifiers.go
+++ b/pkg/resource/web_acl/identifiers.go
@@ -53,3 +53,12 @@ func (ri *resourceIdentifiers) Region() *ackv1alpha1.AWSRegion {
 	}
 	return nil
 }
+
+// Partition returns the AWS partition in which the reosurce exists, or
+// nil if this information is not known.
+func (ri *resourceIdentifiers) Partition() *ackv1alpha1.AWSPartition {
+	if ri.meta != nil {
+		return ri.meta.Partition
+	}
+	return nil
+}

--- a/pkg/resource/web_acl/manager.go
+++ b/pkg/resource/web_acl/manager.go
@@ -75,6 +75,8 @@ type resourceManager struct {
 	awsAccountID ackv1alpha1.AWSAccountID
 	// The AWS Region that this resource manager targets
 	awsRegion ackv1alpha1.AWSRegion
+	// The AWS Partition that this resource manager targets
+	awsPartition ackv1alpha1.AWSPartition
 	// sdk is a pointer to the AWS service API client exposed by the
 	// aws-sdk-go-v2/services/{alias} package.
 	sdkapi *svcsdk.Client
@@ -193,7 +195,8 @@ func (rm *resourceManager) Delete(
 // name for the resource
 func (rm *resourceManager) ARNFromName(name string) string {
 	return fmt.Sprintf(
-		"arn:aws:wafv2:%s:%s:%s",
+		"arn:%s:wafv2:%s:%s:%s",
+		rm.awsPartition,
 		rm.awsRegion,
 		rm.awsAccountID,
 		name,
@@ -373,6 +376,7 @@ func newResourceManager(
 		rr:           rr,
 		awsAccountID: id,
 		awsRegion:    region,
+		awsPartition: ackv1alpha1.AWSPartition(cfg.Partition),
 		sdkapi:       svcsdk.NewFromConfig(clientcfg),
 	}, nil
 }

--- a/pkg/resource/web_acl/sdk.go
+++ b/pkg/resource/web_acl/sdk.go
@@ -6555,6 +6555,9 @@ func (rm *resourceManager) setStatusDefaults(
 	if ko.Status.ACKResourceMetadata.Region == nil {
 		ko.Status.ACKResourceMetadata.Region = &rm.awsRegion
 	}
+	if ko.Status.ACKResourceMetadata.Partition == nil {
+		ko.Status.ACKResourceMetadata.Partition = &rm.awsPartition
+	}
 	if ko.Status.ACKResourceMetadata.OwnerAccountID == nil {
 		ko.Status.ACKResourceMetadata.OwnerAccountID = &rm.awsAccountID
 	}

